UPGRADE-3.1.md

```diff
@@ -24,6 +24,22 @@ FrameworkBundle
 
  * As it was never an officially supported feature, the support for absolute
    template paths has been deprecated and will be removed in Symfony 4.0.
+ * Since the refactoring of 2.8, form types without dependencies are autoloaded
+   by the `DependencyInjectionExtension` in the form registry.
+   Consequently they are deprecated and will be removed in 4.0.
+   You should use the fully-qualified name instead.
+
+   Before:
+
+   ```php
+   $type = $this->container->get('form.type.text');
+   ```
+
+   After:
+
+    ```php
+    $type = '\Symfony\Component\Form\Extension\Core\Type\TextType':
```

Comment:
I would simply list the IDs of the deprecated services. Passing them to the form builder should already produce errors as the builder is expecting a string.